pytest/lib/bridge.py

```diff
@@ -0,0 +1,249 @@
+import atexit
+import base64
+import multiprocessing
+from pathlib import Path
+import signal
+import shutil
+import subprocess
+import time
+import traceback
+import os
+
+def atexit_cleanup(obj):
+    print("Cleaning %s on script exit" % (obj.__class__.__name__))
+    try:
+        obj.cleanup()
+    except:
+        print("Cleaning failed!")
+        traceback.print_exc()
+        pass
+
+class Cleanable(object):
+
+    def kill(self):
+        if self.pid.value != 0:
+            os.kill(self.pid.value, signal.SIGKILL)
+            self.pid.value = 0
+    
+    def cleanup(self):
+        if self.cleaned:
```

Comment:
The idiomatic way of doing this is to have `__init__` in `Cleanable`, which sets `self.cleaned` to `False`, and then in each subclass to call `super().__init__()`, so that subclasses don't have to know / care about setting `cleaned`

Reply:
++